### ISSUE TITLE
sys plugin

### ISSUE THREAD

**Issue by LordMZTE** [rhaiscript/rhai#451]
There currently seems to be no plugin/library that allows giving rhai scripts full system access by allowing them to do things like spawning subprocesses or doing filesystem operations. Is something like this planned as an optional feature or has someone already made this as a seperate crate?

**Comment by schungx** [comment]
Not really.  It was originally planned but nobody sorta got around to it...

**Comment by schungx** [comment]
Seeking contributors to build up Rhai's standard library.

**Comment by cn-kali-team** [comment]
When I chose the script engine, I found that there is a standard module. We can learn from it

[rune_modules](https://docs.rs/rune-modules/latest/rune_modules/)

**Comment by schungx** [comment]
It  is true.  It does leave something to be desired.

I agree that having a basic standard module is a quite urgent need.

**Comment by schungx** [comment]
It  is true.  It does leave something to be desired.

I agree that having a basic standard module is a quite urgent need.

**Comment by naturallymitchell** [comment]
Jazz's standard [fs library](https://github.com/jazzdotdev/jazz/blob/master/src/bindings/system/fs.rs) and [fs-extras library](https://github.com/jazzdotdev/fs-extras) are adequate for Jazz Flute, the standard application framework.

The other crates that are integrated by default in the scripting environment can also be found in the [bindings directory](https://github.com/jazzdotdev/jazz/tree/master/src/bindings).

It looks pretty easy to integrate Jazz and Rhai.

**Comment by schungx** [comment]
@dankope has indicated that he'd work on this... let me see if this mention is enough to assign this issue to him...

**Comment by dankope** [comment]
> EDIT: Hmmm....  looks like no.  @dankope, would you make a comment here so I can assign you?

Should be able to assign me now, hopefully.

**Comment by dankope** [comment]
First package done! Please try it out and leave feedback on what should be changed/fixed, thanks. https://github.com/rhaiscript/rhai-fs

The next package will most likely be networking. The networking package will only use what is available in the rust `std` library to insure compatibility.

**Comment by eliabieri** [comment]
Any updates on the networking package?

**Comment by dankope** [comment]
> Any updates on the networking package?

Hi, thank you for asking, I have been busy lately however I should have some time next week to get it in a good shape for publishing, thanks again for taking the time to ask.

**Comment by dankope** [comment]
The networking package has been pushed to the preliminary stage (aka not in the rhai org yet) here: https://github.com/dankope/rhai-net! Currently it supports TCP server and client functionality however there is quite a bit still yet to be done, I will have the issues for it filled out later tomorrow hopefully, so everyone is on the same page as to what to expect.

**Comment by schungx** [comment]
@eliabieri I wonder if the networking package is still progressing?

**Comment by emesare** [comment]
Hi all, as you can tell by the above comment, I have not had time to continue working on the standard library. 

I am going to unassign myself from the issue (I should have done this years ago)

I really hope someone can pickup this!

**Comment by schungx** [comment]
Done!
